src/wesnoth.cpp

```diff
@@ -1103,33 +1102,23 @@ int main(int argc, char** argv)
 
 		if(terminal_switches.find(arg) != terminal_switches.end() ||
 			std::find_if(terminal_arg_switches.begin(), terminal_arg_switches.end(), switch_matches_arg) != terminal_arg_switches.end()) {
-			terminal_force = true;
-		}
-
-#ifdef _WIN32
-		if(arg == "--wnoconsole") {
-			terminal_force = false;
-		} else if(arg == "--wconsole") {
-			terminal_force = true;
-		} else if(arg == "--wnoredirect") {
 			write_to_log_file = false;
 		}
-#endif
 
 		if(arg == "--no-log-to-file") {
-			terminal_force = true;
+			write_to_log_file = false;
```

Comment:
I think the removal of `terminal_force` looks wrong. If one of the "terminal switches" is found, shouldn't `write_to_log_file` be set to `true` instead of `false` on line 1105?

Reply:
If a terminal switch is found then that would mean it should write output to the terminal, in which case `write_to_log_file` should be false.

Reply:
Oh, right. Never mind then.